packages/transaction-controller/src/TransactionController.ts

```diff
@@ -957,6 +971,53 @@ export class TransactionController extends BaseController<
     }
   }
 
+  /**
+   * Appends new sendFlowHistoryToAdd to the transaction with the given transactionID
+   * if transaction is unapproved. Returns the updated transactionMeta.
+   *
+   * @param transactionID - The ID of the transaction to update.
+   * @param currentSendFlowHistoryLength - The length of the current sendFlowHistory array.
+   * @param sendFlowHistoryToAdd - The sendFlowHistory entries to add.
+   * @returns The updated transactionMeta.
+   */
+  updateTransactionSendFlowHistory(
+    transactionID: string,
+    currentSendFlowHistoryLength: number,
+    sendFlowHistoryToAdd: SendFlowHistoryEntry[],
+  ): TransactionMeta {
+    if (this.isSendFlowHistoryDisabled) {
+      throw new Error(
+        'TransactionsController: Send flow history is disabled for this instance of the controller.',
+      );
+    }
+
+    const transactionMeta = this.getTransaction(transactionID);
+
+    if (!transactionMeta) {
+      throw new Error(
+        `TransactionsController: updateTransactionSendFlowHistory called but transactionMeta not found.`,
+      );
+    }
+
+    validateIfTransactionUnapproved(
```

Comment:
Should we keep this logic inline as this is the only use thus far so we avoid additional public methods we should ideally be unit testing?

Reply:
I've intentionally added this to the utils because it will be used in lot of places in subsequent PRs. 
Also agree that PR's should be handled independently but I fear that will missed easily and duplicated since other features not related with this.

Reply:
Like `getTransaction`. We have that helper but also `.find` calls on `transactions` in some places.

Reply:
Update: I will re-use this helper in https://github.com/MetaMask/MetaMask-planning/issues/1099